examples/bandwidthconstraints/src/index.js

```diff
@@ -0,0 +1,146 @@
+  if (!canvas) {
+    waveformContainer.appendChild(waveform.element);
+  }
+}
+
+/**
+ * Attach a Track to one of the HTMLMediaElements.
+ */
+function attachTrack(audioElement, videoElement, track) {
+  if (track.kind === 'audio') {
+    attachAudioTrack(track, audioElement);
+    return;
+  }
+  track.attach(videoElement);
+}
+
+/**
+ * Detach the AudioTrack from the HTMLAudioElement and stop the Waveform.
+ */
+function detachAudioTrack(track, audioElement) {
+  track.detach(audioElement);
+  waveform.unsetStream();
+  var canvas = waveformContainer.querySelector('canvas');
+  if (canvas) {
+    canvas.remove();
+  }
+}
+
+/**
+ * Detach a Track from its HTMLMediaElement.
+ */
+function detachTrack(audioElement, videoElement, track) {
+  if (track.kind === 'audio') {
+    detachAudioTrack(track, audioElement);
+    return;
+  }
+  track.detach(videoElement);
+}
+
+/**
+ * Connect to or disconnect from a Room.
+ */
+function connectToOrDisconnectFromRoom(e) {
+  e.preventDefault();
+  if (room) {
+    e.target.value = 'Connect to Room';
+    room.disconnect();
+    room = null;
+    return;
+  }
+  var maxAudioBitrate = audioBitrateSelector.value
+    ? Number(audioBitrateSelector.value)
+    : null;
+  var maxVideoBitrate = videoBitrateSelector.value
+    ? Number(videoBitrateSelector.value)
+    : null;
+
+  getRoomCredentials().then(function(creds) {
+    return connectWithBandwidthConstraints(
+      creds.token,
+      roomName,
+      maxAudioBitrate,
+      maxVideoBitrate);
+  }).then(function(_room) {
+    room = _room;
+    e.target.value = 'Disconnect from Room';
+  });
+}
+
+/**
+ * Update bandwidth constraints in the Room.
+ */
+function updateBandwidthParametersInRoom() {
+  if (!room) {
+    return;
+  }
+  var maxAudioBitrate = audioBitrateSelector.value
+    ? Number(audioBitrateSelector.value)
+    : null;
+  var maxVideoBitrate = videoBitrateSelector.value
+    ? Number(videoBitrateSelector.value)
+    : null;
+  updateBandwidthConstraints(room, maxAudioBitrate, maxVideoBitrate);
+}
+
+// Load the code snippet.
+getSnippet('./helpers.js').then(function(snippet) {
+  var pre = document.querySelector('pre.language-javascript');
+  pre.innerHTML = Prism.highlight(snippet, Prism.languages.javascript);
+}).then(function() {
+  // Set listeners to the bandwidth selectors.
+  audioBitrateSelector.onchange = updateBandwidthParametersInRoom;
+  videoBitrateSelector.onchange = updateBandwidthParametersInRoom;
+
+  // Set listener to the connect or disconnect button.
+  connectOrDisconnect.onclick = connectToOrDisconnectFromRoom;
+
+  // Get the credentials to connect to the Room.
+  return getRoomCredentials();
+}).then(function(creds) {
+  // Connect to a random Room with no media. This Participant will
+  // display the media of the second Participant that will enter
+  // the Room with bandwidth constraints.
+  return Video.connect(creds.token, { tracks: [] });
+}).then(function(_room) {
```

Comment:
I think we can use `async`/`await` now. This also lets us avoid binding `_room`.

Reply:
@markandrus Do Firefox and Safari support it?

Reply:
Yes, recent versions support it.